src/ClearingHouse.sol

```diff
@@ -209,24 +212,42 @@ contract ClearingHouse is Policy, RolesConsumer, CoolerCallback {
         fundTime += FUND_CADENCE;
 
         uint256 balance = sDai.maxWithdraw(address(this));
+        uint256 outstandingDebt = TRSRY.reserveDebt(dai, address(this));
         // Rebalance funds on hand with treasury's reserves.
         if (balance < FUND_AMOUNT) {
+            // Since users will take their loans in DAI,
+            // the clearinghouse debt is set in DAI terms.
             uint256 fundAmount = FUND_AMOUNT - balance;
-            // Since TRSRY denominates in sDAI, a conversion must be done beforehand.
+            TRSRY.setDebt({
+                debtor_: address(this),
+                token_: dai,
+                amount_: outstandingDebt + fundAmount
+            });
+
+            // Since TRSRY holds sDAI, a conversion must be done before
+            // funding the clearinghouse.
             uint256 amount = sDai.previewWithdraw(fundAmount);
-            // Fund the clearinghouse with treasury assets.
-            TRSRY.increaseDebtorApproval(address(this), sDai, amount);
-            TRSRY.incurDebt(sDai, amount);
-            // Sweep into DSR if necessary
+            TRSRY.increaseWithdrawApproval(address(this), sDai, amount);
+            TRSRY.withdrawReserves(address(this), sDai, amount);
```

Comment:
Won't this result in double-counting of debt?

`fundAmount` is being added wrt the DAI debt
`amount` (`fundAmount` converted to sDAI terms) of sDAI withdrawn from TRSRY, which would increase the sDAI debt

Reply:
TRSRY allows for 2 different types of interactions:
- debt functions: allow an address to borrow the funds (log deb + transfer funds)
- withdraw functions: allow an address to withdraw the funds (no deb is logged)

so in this case, we use `setDebt(DAI)` because the clearinghouse debt is denominated in DAI, but since TRSRY will hold  sDAI, we have to withdraw the equivalent sDAI amount.

for reference: https://github.com/OlympusDAO/olympus-v3/blob/master/src/modules/TRSRY/OlympusTreasury.sol